### PR TITLE
[RSDK-8266]   Refactor OAK models and support new FFC modular config

### PR DESCRIPTION
**Pull request by hexbabe**
[RSDK-8266](https://viam.atlassian.net/browse/RSDK-8266)

This PR:
- Changes file structure to be more robust as module codebase grows in size
- Removes `viam:camera:oak-ffc` from the `meta.json` (I re-ran `viam module update` and its gone from the registry too; no user usages of it so we're chilling)
- Adds new models to code only (not `meta.json`) `viam:luxonis:oak-d` and `viam:luxonis:oak-ffc-3p`
- Adds deprecation warning for old models `viam:camera:oak` and `viam:camera:oak-d`. We are keeping the old deprecated models for now, will prob do migration later
- Adds new validation flow using a `Validator` class
- Adds unit tests for new validation flow

This PR does not:
- Implement functionality for the OAK-FFC-3P
- Add anything new outside of validation flow for OAK-D and new OAK-FFC-3P config outlined in the scope doc for customer

### Testing
```
{
  "sensors": [
    "color",
    "depth"
  ],
  "height_px": 800,
  "width_px": 800
}
```
Regression tested OAK-D
- Initializes properly ✅
- Reconfigures properly (changing resolution, fps, sensors) ✅
- get_image works for color and depth ✅
- get_point_cloud works ✅
- Unplug, replug camera works ✅

[RSDK-8266]: https://viam.atlassian.net/browse/RSDK-8266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/oak.py [hunk 1]
```diff
@@ -91,12 +87,24 @@ class Properties(NamedTuple):
...
 
-    family = ModelFamily("viam", "camera")
-    MODELS: ClassVar[Tuple[Model]] = (
-        Model(family, "oak"),
-        Model(family, "oak-ffc"),
-        Model(family, "oak-d"),
+    _deprecated_family = ModelFamily("viam", "camera")
+    _depr_oak_agnostic_model = Model(_deprecated_family, "oak")
+    _depr_oak_d_model = Model(_deprecated_family, "oak-d")
+    DEPRECATED_MODELS: ClassVar[Tuple[Model]] = (
```
Comment (seanavery) [inline src/oak.py:93]: [q] Continue to register deprecated models for update log [here](https://github.com/viamrobotics/viam-camera-oak/pull/48/files#diff-491f430903c5a126fd9e3f25477482a4133b3c896aefac40746ac2cc802eeddcR157)?

File: src/helpers/validators.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+from google.protobuf.struct_pb2 import Struct
+
+from viam.errors import ValidationError
+from viam.proto.app.robot import ComponentConfig
+
+
+from src.worker.worker import Worker
+
+
+class Validator:
```
Comment (seanavery) [inline src/helpers/validators.py:13]: Any significant update to the helpers here or just moved over?

**Review by seanavery — APPROVED** [review]
LGTM.

Awesome that `viam module update` removes deprecated model in registry 🎉

File: src/oak.py (hunk 1 above)
Comment (hexbabe) [reply src/oak.py:93]: Added to a bullet

File: src/helpers/validators.py (hunk 2 above)
Comment (hexbabe) [reply src/helpers/validators.py:13]: Validators is new. The existing code that was moved was from the validate function. Now there's no real logic there for actual validations. It all takes place in `validators.py`

`helpers.py` just moves existing shared code into a single file.

**Review by randhid — APPROVED** [review]
LGTM, nice tests.

File: src/helpers/validators.py (hunk 2 above)
Comment (seanavery) [reply src/helpers/validators.py:13]: sounds good

**Pull request merged**